test-monitor/nitestmonitor.yml

```diff
@@ -632,6 +632,26 @@ definitions:
         type: array
         items:
           $ref: '#/definitions/ResultSortDefinitionObject'
+  AdvancedQuery:
+    title: Advanced Query Object for Test Data
+    type: object
+    properties:
+      filter:
+        description: The query filter, in Dynamic Linq
```

Comment:
What we did, for the asset filtering APIs, is enumerate all properties that can be used in the filter. I think this is important as we need to document all possible/supported properties which can participate in filtering. 
You can also take the introductory query text we have written for the query filter. take a look at the model of this route: http://demo.systemlink.io/niapis/?urls.primaryName=Asset%20Performance%20Management#/assets/query-assets

This way all our APIs could be aligned with regards to the documentation we provide for our filters.

Reply:
After some discussion, our plan is to create documentation for the query language in Markdown, which will live in this repo next to the swagger doc. The documentation in the swagger doc itself will contain a simple example, plus a link to the markdown document here for further info. The contents of the markdown document will be adapted from the query text linked above, as well as the existing documentation for the query language.

We have a separate work item to track this documentation, so it won't be included as part of this review.